src/Common/src/TypeSystem/IL/McgInteropSupport.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Internal.TypeSystem;
+using Debug = System.Diagnostics.Debug;
+
+using System.Reflection;
+
+namespace Internal.IL
+{
+    /// <summary>
+    /// Provides compilation hooks for interop code generated by Mcg tool
+    /// </summary>
+    public static class McgInteropSupport
+    {
+        /// <summary>
+        /// Assembly name suffix for Mcg generated interop code.
+        /// </summary>
+        private const string AssemblyNameSuffix = ".McgInterop";
+
+        /// <summary>
+        /// Returns true if <paramref name="method"/> is Mcg generated PInvoke
+        /// </summary>
+        public static bool IsPregeneratedPInvoke(MethodDesc method)
+        {
+            Debug.Assert(method.IsPInvoke);
+
+            var metadataType = (MetadataType)method.OwningType;
+            var module = metadataType.Module;
+
+            var assemblyName = ((IAssemblyDesc)module).GetName();
+            var simpleName = assemblyName.Name;
+
+            return simpleName.EndsWith(AssemblyNameSuffix);
+        }
+
+        /// <summary>
+        /// Returns pregenerated interop code for given PInvoke method if one exist
+        /// </summary>
+        public static MethodDesc TryGetPregeneratedPInvoke(MethodDesc method)
+        {
+            Debug.Assert(method.IsPInvoke);
+
+            var metadataType = (MetadataType)method.OwningType;
+            var module = metadataType.Module;
+
+            var assemblyName = ((IAssemblyDesc)module).GetName();
+
+            var interopAssemblyName = new AssemblyName();
+
+            interopAssemblyName.Name = assemblyName.Name + AssemblyNameSuffix;
+            interopAssemblyName.Version = assemblyName.Version;
+            interopAssemblyName.SetPublicKeyToken(interopAssemblyName.GetPublicKeyToken());
+            interopAssemblyName.CultureName = assemblyName.CultureName;
+            interopAssemblyName.ContentType = assemblyName.ContentType;
+
+            var interopModule = module.Context.ResolveAssembly(interopAssemblyName, false);
+            if (interopModule == null)
+                return null;
+
+            var generated = GetMatchingMethod(interopModule, method);
+            Debug.Assert(generated != null, "Missing method in .McgInterop assembly" + method.ToString());
```

Comment:
I made it a hard error. It turns out to be a problem it can be always converted to warning.